src/backend/parser/parse_relation.c

```diff
@@ -1355,7 +1355,8 @@ addRangeTableEntryForFunction(ParseState *pstate,
 		 * mark this here because this is where we know that the function is being
 		 * used as a RangeTableEntry.
 		 */
-		if (funcexpr && IsA(funcexpr, FuncExpr))
+		Assert(funcexpr);
```

Comment:
Pointer funcexpr cannot be NULL here, because it may be dereferenced [below](https://github.com/arenadata/gpdb/pull/748/files#diff-d35191b290e0e158e14a958e47730fb02e03f30c0153a608b13d90606b9d681bR1423)

Reply:
Why won't we keep `IsA`? Postgres constantly checks that returned node type is correct

Reply:
> Why won't we keep `IsA`?

What does it mean we won’t save `IsA`?

Reply:
I mean you removed `IsA(funcexpr, FuncExpr)` check, why?

Reply:
I did not

> removed `IsA(funcexpr, FuncExpr)` check

!

Reply:
Sorry, my bad